### PR TITLE
Support explicit type and name during table creation

### LINKED ISSUE #10207: Create `Struct` table with explicit type and name
### Is your feature request related to a problem or challenge?

Currently we can build struct table like this

```
statement ok
create table t as values (struct(1, 'a')), (struct(2, 'b'));

query ?
select * from t;
----
{c0: 1, c1: a}
{c0: 2, c1: b}
```

If we want to name them, we need to build with `named_struct`

```
statement ok
create table t as values (named_struct('number', 1, 'varchar', 'a')), (named_struct('number', 2, 'varchar', 'b'));

query ?
select * from t;
----
{number: 1, varchar: a}
{number: 2, varchar: b}
```

I hope we can build the struct table with pre-defined type and name

```
statement error DataFusion error: This feature is not implemented: Unsupported SQL type Custom\(ObjectName\(\[Ident \{ value: "struct", quote_style: None \}\]\), \["n", "int", "s", "varchar"\]\)
create table t (s struct(n int, s varchar)) as values (struct(1, 'a')), (struct(2, 'b'));

query ?
select * from t;
----
{n: 1, s: a}
{n: 2, s: b}
```

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

Similar to DuckDB
https://duckdb.org/docs/sql/data_types/struct#creating-structs-with-the-row-function

### PR DESCRIPTION
**Pull request by duongcongtoai**
## Which issue does this PR close?

Closes #10207 .

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -103,7 +103,7 @@ parquet = { version = "51.0.0", default-features = false, features = ["arrow", "
 rand = "0.8"
 rstest = "0.19.0"
 serde_json = "1"
-sqlparser = { version = "0.45.0", features = ["visitor"] }
+sqlparser = { git = "https://github.com/duongcongtoai/sqlparser-rs", rev = "62f1e02f4a7d5160e8538b34f30e311905b4a72a", features = ["visitor"] }
```
Comment (duongcongtoai) [inline Cargo.toml:106]: temp commithash

**Comment by duongcongtoai** [comment]
waiting for sql-parser upgrade in this PR: https://github.com/apache/datafusion/pull/10392

**Comment by duongcongtoai** [comment]
hi @jayzhan211 the PR is ready for review https://github.com/apache/datafusion/pull/10392, but the syntax is bigquery-liked. Do you need the syntax of duckdb to complete this issue?

**Comment by jayzhan211** [comment]
> hi @jayzhan211 the PR is ready for review, but the syntax is bigquery-liked. Do you need the syntax of duckdb to complete this issue?

It would be nice if we support duckdb-like syntax, we usually follow what postgres and duckdb are, unless there is good reason that we should have better alternative

**Comment by duongcongtoai** [comment]
ok, i'm working on it

**Review by alamb — APPROVED** [review]
Thank you for the contribution @duongcongtoai  -- this looks great to me 🙏

**Pull request merged**